velox/dwio/parquet/reader/ParquetData.cpp

```diff
@@ -82,6 +82,34 @@ bool ParquetData::rowGroupMatches(
   return true;
 }
 
+bool ParquetData::preDecompRowGroup(uint32_t index) {
+#ifdef VELOX_ENABLE_QPL
+  if (!dwio::common::QplJobHWPool::GetInstance().job_ready() ||
+      !needPreDecomp) {
+    LOG(WARNING) << "QPL Job not ready or zlib window size(" << needPreDecomp
+                 << ") is not 4KB";
+    return false;
+  }
+
+  auto& chunk = rowGroups_[index].columns[type_->column];
+  auto& metaData = chunk.meta_data;
+  if (metaData.codec == thrift::CompressionCodec::GZIP) {
```

Comment:
GZIP has window size 15, you are requiring 12 above.  Please create a dedicated enum for this in parquet first.

Reply:
add a new window size 12 to the existed window bits in class Compressor